controllers/etcd_controller.go

```diff
@@ -789,16 +799,14 @@ func (r *EtcdReconciler) addFinalizersToDependantSecrets(etcd *druidv1alpha1.Etc
 }
 
 func (r *EtcdReconciler) removeFinalizersToDependantSecrets(etcd *druidv1alpha1.Etcd) error {
-	var err error
-	storeSecret := corev1.Secret{}
-	err = r.Client.Get(context.TODO(), types.NamespacedName{
-		Name:      etcd.Spec.Backup.Store.SecretRef.Name,
-		Namespace: etcd.Namespace,
-	}, &storeSecret)
-	if err != nil && !errors.IsNotFound(err) {
-		return err
-	}
-	if err == nil {
+	if etcd.Spec.Backup.Store != nil {
+		storeSecret := corev1.Secret{}
+		if err := r.Client.Get(context.TODO(), types.NamespacedName{
+			Name:      etcd.Spec.Backup.Store.SecretRef.Name,
+			Namespace: etcd.Namespace,
+		}, &storeSecret); err != nil {
```

Comment:
It should return ` nil`  is err is err is 'NotFound.'

Reply:
I cannot return nil here if its not found as it will not remove the remaining finalizers.

Reply:
Ahh. Right. But you will have to add check to ignore not found error.